crates/ruff_linter/src/message/diff.rs

```diff
@@ -36,6 +36,16 @@ impl<'a> Diff<'a> {
 
 impl Display for Diff<'_> {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        // TODO: Instead of hard-coding this in, the `TextEmitter` should be
+        // handling it. See https://github.com/astral-sh/ruff/issues/12597 for
+        // further discussion.
+        if matches!(
+            self.fix.applicability(),
+            Applicability::Unsafe | Applicability::DisplayOnly
+        ) {
+            return Ok(());
+        }
```

Comment:
Does this mean that we won't see unsafe fixes in snapshot tests anymore?

I would find this concerning because it means we loose all test coverage for unsafe fixes (of which there are plenty)

Reply:
Yes, we need to address this before merging.

Reply:
Yeah, we realized this late in our time working on it yesterday. It may warrant going ahead and doing the bit I highlighted in #12597, etc. I think this was a case where trying to keep the change minimal actually makes it harder to land—this particularly bit of coupling wasn’t obvious to me until the very end when we changed from including all the CTAs to including *none* of them, and turning off.

I think it probably needs the aforementioned refactor (where the `TextEmitter` handles what to print or not) plus setting the relevant flags based on both `--unsafe-fixes` *and* whether it is testing.